codegen-sbt/src/main/scala/caliban/codegen/CalibanSourceGenerator.scala

```diff
@@ -94,25 +96,31 @@ object CalibanSourceGenerator {
           files           <- Codegen.generate(opts, settings.genType).asSomeError
         } yield files
 
-      Runtime.default
-        .unsafeRun(
-          for {
-            fromFiles <- ZIO.foreach(sources.toList)(source =>
-                           generateFileSource(source, collectSettingsFor(fileSettings, source).settings).catchAll {
-                             case Some(reason) =>
-                               putStrLn(reason.toString) *> putStrLn(reason.getStackTrace.mkString("\n")).as(List.empty)
-                             case None         => ZIO.succeed(List.empty)
-                           }
-                         )
-            fromUrls  <- ZIO.foreach(urlSettings)(setting =>
-                           generateUrlSource(setting.url, setting.settings).catchAll {
-                             case Some(reason) =>
-                               putStrLn(reason.toString) *> putStrLn(reason.getStackTrace.mkString("\n")).as(List.empty)
-                             case None         => ZIO.succeed(List.empty)
-                           }
-                         )
-          } yield (fromFiles ++ fromUrls).flatten
-        )
+      val generateFromFiles = ZIO
+        .foreach(sources.toList) { source =>
+          ZIO
+            .collectAll(
+              collectSettingsFor(fileSettings, source).map(s => generateFileSource(source, s.settings))
+            )
+            .catchAll {
+              case Some(reason) =>
+                putStrLn(reason.toString) *> putStrLn(reason.getStackTrace.mkString("\n")).as(List.empty)
+              case None         => ZIO.succeed(List.empty)
+            }
+            .map(_.flatten)
+        }
+
+      val generateFromURLs = ZIO.foreach(urlSettings)(setting =>
+        generateUrlSource(setting.url, setting.settings).catchAll {
+          case Some(reason) =>
+            putStrLn(reason.toString) *> putStrLn(reason.getStackTrace.mkString("\n")).as(List.empty)
+          case None         => ZIO.succeed(List.empty)
+        }
+      )
+
+      Runtime.default.unsafeRun {
+        ZIO.mapN(generateFromFiles, generateFromURLs)((_ ++ _)).map(_.flatten)
```

Comment:
Can `parMapN` speed up things in this case? 🤔

Reply:
IME they're so fast there's no need for this. I'd rather have sbt manage parallelism on a higher level (more tasks at once or whatever), but I'm fine with changing this if you prefer

Reply:
I don't have an opinion. Was just asking 🙂

Reply:
cool, I'd leave it as it is for now then :)